library/std/src/sys/mod.rs

```diff
@@ -76,3 +76,49 @@ cfg_if::cfg_if! {
         pub mod c;
     }
 }
+
+#[cfg(not(test))]
+cfg_if::cfg_if! {
+    if #[cfg(target_os = "android")] {
+        pub use crate::android::log2f32;
+        pub use crate::android::log2f64;
+    } else {
+        pub fn log2f32(n: f32) -> f32 {
+            unsafe { crate::intrinsics::log2f32(n) }
+        }
+
+        pub fn log2f64(n: f64) -> f64 {
+            unsafe { crate::intrinsics::log2f64(n) }
+        }
+    }
+}
+
+// Solaris/Illumos requires a wrapper around log, log2, and log10 functions
+// because of their non-standard behavior (e.g., log(-n) returns -Inf instead
+// of expected NaN).
+#[cfg(not(test))]
+cfg_if::cfg_if! {
+    if #[cfg(any(target_os = "solaris", target_os = "illumos"))] {
+        pub fn log_wrapper<F: Fn(f64) -> f64>(n: f64, log_fn: F) -> f64 {
```

Comment:
I'd prefer that we keep functions (certainly ones that are more than one-liners) outside of `cfg_if!` bodies, since those don't get formatting and IDEs have trouble with them.